pegasus_wrapper/workflow.py

```diff
@@ -311,6 +400,288 @@ def _run_python_job(
+        post_job_bash = "\n".join(
+            chain(
+                [
+                    f"cp {str(o_file.scratch.absolute())} {str(o_file.nas.absolute())}"
+                    for o_file in converted_output_files.values()
+                ],
+                post_docker_bash,
+            )
+        )
+
+        # Generate the command to run the python job
+        python_start = (
+            f"-m {python_module_or_path_on_docker}"
+            if isinstance(python_module_or_path_on_docker, str)
+            else str(python_module_or_path_on_docker)
+        )
+        docker_run_command = (
+            f"{python_executable_path_in_docker} {python_start} {python_args}"
+        )
+
+        return self.run_container(
+            job_name,
+            container.name,
+            modified_docker_args,
+            docker_run_command,
+            container.image,
+            depends_on=depends_on,
+            job_is_stageable=job_is_stageable,
+            job_bypass_staging=job_bypass_staging,
+            times_to_retry_job=times_to_retry_job,
+            job_profiles=job_profiles,
+            pre_job_bash=pre_job_bash,
+            post_job_bash=post_job_bash,
+            category=category,
+            resource_request=resource_request,
+        )
+
+    def run_container(
+        self,
+        job_name: Locator,
+        docker_image_name: str,
+        docker_args: str,
+        docker_run_comand: str,
+        docker_tar_path: str,
+        *,
+        depends_on,
+        resource_request: Optional[ResourceRequest] = None,
+        category: Optional[str] = None,
+        pre_job_bash: str = "",
+        post_job_bash: str = "",
+        job_is_stageable: bool = False,
+        job_bypass_staging: bool = True,
+        times_to_retry_job: int = 0,
+        job_profiles: Optional[List[PegasusProfile]] = None,
+    ) -> DependencyNode:
+
+        job_dir = self.directory_for(job_name)
+        ckpt_name = job_name / "___ckpt"
+        checkpoint_path = job_dir / "___ckpt"
+        depends_on = _canonicalize_depends_on(depends_on)
+
+        signature = (docker_image_name, docker_args)
+        if signature in self._signature_to_job:
+            logging.info("Job %s recognized as a duplicate", job_name)
+            return self._signature_to_job[signature]
+
+        script_path = job_dir / "___run.sh"
+
+        # Part of one strategy to run a container through a bash script
+        self._docker_script_generator.write_shell_script_to(
+            docker_image_name=docker_image_name,
+            docker_command=docker_run_comand,
+            docker_tar_path=docker_tar_path,
+            working_directory=job_dir,
+            script_path=script_path,
+            cmd_args=docker_args,
+            ckpt_path=checkpoint_path,
+            pre_job=pre_job_bash,
+            post_job=post_job_bash,
+        )
+
+        # TODO - Refactor this so it uses the BASH transformation to form a job
```

Comment:
Is there an issue created for this? You can then link the issue number from this TODO.